.editorconfig

```diff
@@ -6,14 +6,297 @@
+dotnet_naming_rule.locals_should_be_camel_case.symbols = locals_and_parameters
+dotnet_naming_rule.locals_should_be_camel_case.style = camel_case_style
+
+dotnet_naming_symbols.locals_and_parameters.applicable_kinds = parameter, local
+
+dotnet_naming_style.camel_case_style.capitalization = camel_case
+
+# Local functions are PascalCase
+dotnet_naming_rule.local_functions_should_be_pascal_case.severity = suggestion
+dotnet_naming_rule.local_functions_should_be_pascal_case.symbols = local_functions
+dotnet_naming_rule.local_functions_should_be_pascal_case.style = local_function_style
+
+dotnet_naming_symbols.local_functions.applicable_kinds = local_function
+
+dotnet_naming_style.local_function_style.capitalization = pascal_case
+
+# By default, name items with PascalCase
+dotnet_naming_rule.members_should_be_pascal_case.severity = suggestion
+dotnet_naming_rule.members_should_be_pascal_case.symbols = all_members
+dotnet_naming_rule.members_should_be_pascal_case.style = pascal_case_style
+
+dotnet_naming_symbols.all_members.applicable_kinds = *
+
+dotnet_naming_style.pascal_case_style.capitalization = pascal_case
+
+file_header_template = Copyright (C) 2015-2024 The Neo Project.\n\n{fileName} file belongs to the neo project and is free\nsoftware distributed under the MIT software license, see the\naccompanying file LICENSE in the main directory of the\nrepository or http://www.opensource.org/licenses/mit-license.php\nfor more details.\n\nRedistribution and use in source and binary forms with or without\nmodifications are permitted.
+
+# Require file header
+dotnet_diagnostic.IDE0073.severity = error
+
+# RS0016: Only enable if API files are present
+dotnet_public_api_analyzer.require_api_files = true
+
+# IDE0055: Fix formatting
+# Workaround for https://github.com/dotnet/roslyn/issues/70570
+dotnet_diagnostic.IDE0055.severity = warning
+
+# CSharp code style settings:
+[*.cs]
+# Newline settings
+csharp_new_line_before_open_brace = all
+csharp_new_line_before_else = true
+csharp_new_line_before_catch = true
+csharp_new_line_before_finally = true
+csharp_new_line_before_members_in_object_initializers = true
+csharp_new_line_before_members_in_anonymous_types = true
+csharp_new_line_between_query_expression_clauses = true
+
+# Indentation preferences
+csharp_indent_block_contents = true
+csharp_indent_braces = false
+csharp_indent_case_contents = true
+csharp_indent_case_contents_when_block = true
+csharp_indent_switch_labels = true
+csharp_indent_labels = flush_left
+
+# Whitespace options
+csharp_style_allow_embedded_statements_on_same_line_experimental = false
+csharp_style_allow_blank_lines_between_consecutive_braces_experimental = false
+csharp_style_allow_blank_line_after_colon_in_constructor_initializer_experimental = false
+csharp_style_allow_blank_line_after_token_in_conditional_expression_experimental = false
+csharp_style_allow_blank_line_after_token_in_arrow_expression_clause_experimental = false
+
+# Prefer "var" everywhere
```

Comment:
Personally my opinion is that `var` obscures the actual variable type. Using var makes it hard to know what type the underlying variable actually is. A properly named variable speaks for itself.

So, can we vote about remove `var` ?

Reply:
Not a good idea, var exists for a reason which is make the coding easier, all IDEs can show the type. Using `var` or something a like is the trend of programming languages, including C++ also added `auto`.

Reply:
And another issue is that we may not really remember what the type is, but we can still write code, IDE will tell us the type. If you ban var, every time we want to define a variable, we must know the exact type before we can go on with the processing logic, types can be:

Stack<List<ILocalSymbol>>
Dictionary<ILocalSymbol, byte>
Dictionary<ILabelSymbol, JumpTarget>
Dictionary<IParameterSymbol, byte>

Reply:
`Dictionary<ILocalSymbol, byte> s=new()` or `var s = new Dictionary<ILocalSymbol, byte>()`

Reply:
> `Dictionary<ILocalSymbol, byte> s=new()` or `var s = new Dictionary<ILocalSymbol, byte>()`

I prefer the first one, cause you are defining it directly from the type:

what about this one 

```C#
AttributeData? initialValue = field.GetAttributes().FirstOrDefault(p => p.AttributeClass!.Name == nameof(InitialValueAttribute))
```
how about this one
```C#
ILocalSymbol[] symbols = ((ParenthesizedVariableDesignationSyntax)((DeclarationExpressionSyntax)syntax.Variable).Designation).Variables.Select(p => (ILocalSymbol)model.GetDeclaredSymbol(p)!).ToArray();
```
and this one:
```C#
var sections = syntax.Sections.Select(p => (p.Labels, p.Statements, Target: new JumpTarget())).ToArray();
```
and this one:
```C#
            var namedArguments = arguments.OfType<ArgumentSyntax>().Where(p => p.NameColon is not null).Select(p => (Symbol: (IParameterSymbol)model.GetSymbolInfo(p.NameColon!.Name).Symbol!, p.Expression)).ToDictionary(p => p.Symbol, p => p.Expression, (IEqualityComparer<IParameterSymbol>)SymbolEqualityComparer.Default);
```

Reply:
Are you sure you already know the type before you actually write the code?

Reply:
take a guess, what is the type of `symbol.ContainingType.TypeKind`,,, give it a type before you actually access TypeKind, with var, you can just forget the type for now, cause the IDE will tell you later, but without var, you will have to stop there and check the type.

Reply:
> I prefer the first one, cause you are defining it directly from the type:

Me too, but this require to avoid the `var` use

Reply:
Because of this you want to ban `var`? What about the questions that I have above? How do you write code when the type is not clear to you, this is a very very normal case casue you are using libs of others.

Reply:
>Personally my opinion is that var obscures the actual variable type. Using var makes it hard to know what type the underlying variable actually is. A properly named variable speaks for itself.

Standard is not to use it anymore. If it's a problem than you should be naming your variable's name correctly. Example: if you have a `string` you would prefix `str` on the name of the variable, `integers` would be prefix `i`, `booleans` with prefix `b`, and so on. But this is **only** for local variables. 

see https://learn.microsoft.com/en-us/dotnet/standard/design-guidelines/general-naming-conventions

Or if you want even more details see https://learn.microsoft.com/en-us/dotnet/standard/design-guidelines/naming-guidelines

Reply:
> How do you write code when the type is not clear to you

Never happened to me xD

Reply:
> > Personally my opinion is that var obscures the actual variable type. Using var makes it hard to know what type the underlying variable actually is. A properly named variable speaks for itself.
> 
> Standard is not to use it anymore. If it's a problem than you should be naming your variable's name correctly. Example: if you have a `string` you would prefix `str` on the name of the variable, `integers` would be prefix `i`, `booleans` with prefix `b`, and so on. But this is **only** for local variables.
> 
> see https://learn.microsoft.com/en-us/dotnet/standard/design-guidelines/general-naming-conventions
> 
> Or if you want even more details see https://learn.microsoft.com/en-us/dotnet/standard/design-guidelines/naming-guidelines
> 
> Plus you don't need the know the type. You have intellisense, the `IDE` handles all that for you.

So you try to know the type by the name, instead by the type, you need to know the type, and because you want to use var, you need to find a different way

Reply:
I think that we should vote @neo-project/core @AnnaShaleva @roman-khimov, 

- thumb up 👍 for avoid `var`
- thumb down 👎 for `var` anywhere

Reply:
A step that makes the coding much harder. A step that goes against the C# programming habits. A step that you can not answer my questions. For what? For you do not like it? Why dont you focus on some thing much more meaningful, how about add more unit tests, fix issues that are there for years. Banning `var` solves no core issues at all.

Reply:
>So you try to know the type by the name, instead by the type, you need to know the type, and because you want to use var, you need to find a different way

Why do you need to know the type? Just hover over it, use intellisense. We're not all stupid here, remember that. 99.999999% of people program this way in `C#`. Its like you asking to use `struct` instead of a `class`. It doesn't work like that anymore. Its inefficient, slows you down and less lines of code. May I go on? It wouldn't be created if there wasn't a demand for it, just remember that.

If you have type problems, than hover over it.
![image](https://github.com/neo-project/neo/assets/8141309/996609c4-b185-41af-900d-b9c0a73e7ecc)

Just look at core with network protocol classes with these `abstract` inheritances, it's a mess. Also very confusing and difficult to follow. That's why we use `Interfaces` today. `abstract` classes are used when interfaces can't solve the problem. Use the right tools for the right job!

Reply:
> A step that you can not answer my questions. For what? For you do not like it?

What question? I'm asking about a vote, you're trying to impose your criteria, I'm not, so please...

> Why dont you focus on some thing much more meaningful

You opened this PR, and I am reviewing it, here it is defined how to proceed with the codding style from now on, so it does not seem insignificant to me.

Reply:
Review with your professional judgement, with a solid reason, instead of "I like it" or "I dont like it".  C# has `var` for a good reason, but your bannig it has not. You are calling a vote on something of your personal preference.